lib/client/cluster_client.go

```diff
@@ -407,6 +408,11 @@ func (c *ClusterClient) prepareUserCertsRequest(ctx context.Context, params Reis
 		tlsAttestationStatement = tlsSubjectKey.GetAttestationStatement()
 	}
 
+	purpose := proto.UserCertsRequest_CERT_PURPOSE_UNSPECIFIED
+	if params.ReusableMFAResponse != nil {
+		purpose = proto.UserCertsRequest_CERT_PURPOSE_SINGLE_USE_CERTS
+	}
```

Comment:
This runs on client side; the server-side check:

```
	if req.RequesterName == authpb.UserCertsRequest_TSH_DB_EXEC {
		if req.Usage != authpb.UserCertsRequest_Database {
			return trace.BadParameter("requester %s can only request database certificates", req.RequesterName)
		}
		if req.MFAResponse != nil && req.Purpose != authpb.UserCertsRequest_CERT_PURPOSE_SINGLE_USE_CERTS {
			return trace.BadParameter("requester %q can only request single use certificates", req.RequesterName)
		}
	}
```

is only run for `req.RequesterName == authpb.UserCertsRequest_TSH_DB_EXEC`; so what happens if we don't set this requester?

Reply:
If requester is not set, or set to other value, it will go through existing logic where TTL may be adjusted for certain requesters and MFA validation is against `ChallengeAllowReuse_CHALLENGE_ALLOW_REUSE_NO`. 

I could relax this on the server side though, like instead of requesting the `UserCertsRequest_CERT_PURPOSE_SINGLE_USE_CERTS` for purpose from client, we automatically set `UserCertsRequest_CERT_PURPOSE_SINGLE_USE_CERTS` for purpose for `UserCertsRequest_TSH_DB_EXEC`. but i initially did that so it's more explicit and just let the client do more work